eth/ethutils/receipt.go

```diff
@@ -27,6 +27,8 @@ func MarshalReceipt(
 		if t.Protected() {
 			chainId = types.DeriveChainId(&t.V).ToBig()
 		}
+	case *types.DepositTx:
+		// Deposit TX does not have chain ID
```

Comment:
Can we make sure that `chainId` is not used when tx type is depositTx? 
```go
	if signed && chainId != nil {
		signer := types.LatestSignerForChainID(chainId)
		from, _ = txn.Sender(*signer)
	}
```

Reply:
applied!

Reply:
This change was reverted. it does not cover non-protected legacy TX